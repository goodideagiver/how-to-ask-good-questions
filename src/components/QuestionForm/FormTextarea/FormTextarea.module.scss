.root {
	display: grid;

	gap: 1rem;
	align-items: center;
	> * {
		flex: 1;
	}
}

.label {
	font-size: 1.5rem;
	text-align: start;
}

.textarea {
	font-size: 1rem;
	padding: 1rem;
	resize: vertical;
	max-height: 150px;
	min-height: 50px;
	height: fit-content;
	background-color: #424549;
	border: 0;
}

.animate {
	animation: animate 0.3s ease-in-out;
}

@keyframes animate {
	from {
		opacity: 0;
		transform: translate3d(0, -10px, 0);
	}
	to {
		transform: translate3d(0, 0, 0);
	}
}
