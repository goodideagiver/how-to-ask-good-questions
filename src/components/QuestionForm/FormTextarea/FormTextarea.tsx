import classes from './FormTextarea.module.scss';

import { useId } from 'react';

type Props = {
	placeholder: string;
	onChange: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
	value: string;
	label: string;
};

export const FormTextarea = ({
	onChange,
	placeholder,
	value,
	label,
}: Props) => {
	const inputId = useId();

	return (
		<div className={classes.root}>
			<label className={classes.label} htmlFor={inputId}>
				{label}
			</label>
			<textarea
				className={classes.textarea}
				onChange={onChange}
				placeholder={placeholder}
				value={value}
				id={inputId}
			/>
		</div>
	);
};
