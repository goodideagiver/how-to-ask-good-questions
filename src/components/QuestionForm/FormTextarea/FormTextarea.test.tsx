import { getByText, render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { FormTextarea } from './FormTextarea';

describe('FormTextarea', () => {
	it('should have initial value', () => {
		const { container } = render(
			<FormTextarea
				label='asd'
				objectKey='asd'
				onChange={() => void 0}
				placeholder='siem'
				value='test value'
			/>
		);
		expect(getByText(container, 'test value')).toBeInTheDocument();
	});

	it('should have label with test content', () => {
		const { getByText } = render(
			<FormTextarea
				label='test label'
				objectKey='asd'
				onChange={() => void 0}
				placeholder='siem'
				value='qwe'
			/>
		);
		const textareaLabel = getByText('test label');

		expect(textareaLabel).toBeInTheDocument();
		expect(textareaLabel).toBeInstanceOf(HTMLLabelElement);
	});

	it('should have textarea with test content', () => {
		const { getByText } = render(
			<FormTextarea
				label='test label'
				objectKey='asd'
				onChange={() => void 0}
				placeholder='test placeholder'
				value='test value'
			/>
		);
		const textarea = getByText('test value');

		expect(textarea).toBeInTheDocument();
		expect(textarea).toBeInstanceOf(HTMLTextAreaElement);
	});
});
