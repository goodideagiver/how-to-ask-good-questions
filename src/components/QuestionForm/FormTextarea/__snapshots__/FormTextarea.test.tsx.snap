// Vitest Snapshot v1

exports[`FormTextarea > should render 1`] = `
<div>
  <div
    class="_root_83a020"
  >
    <label
      class="_label_83a020"
      for=":r0:"
    >
      asd
    </label>
    <textarea
      class="_textarea_83a020"
      id=":r0:"
      placeholder="siem"
    >
      qwe
    </textarea>
  </div>
</div>
`;
